### PR TITLE
[SPARK-11631][Scheduler] Adding 'Starting DAGScheduler' log

### PR DESCRIPTION
**Pull request by xguo27**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -506,6 +506,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     val (sched, ts) = SparkContext.createTaskScheduler(this, master)
     _schedulerBackend = sched
     _taskScheduler = ts
+    logDebug("Starting DAGScheduler")
```
Comment (jaceklaskowski) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: I'd rather move it to DAGScheduler's constructor as that's where the message belongs.

Comment (xguo27) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: Hi Jacek:

My only concern with putting log in DAGScheduler's constructor is that the logger might not have been initialized when constructor is called.

**Comment by andrewor14** [comment]
Meh, I don't really think this is worth doing. It doesn't add any value or fix anything.

**Comment by xguo27** [comment]
I agree it is trivial, just thought I could quickly add a log statement. If Jacek agrees, I can close those PR.

**Comment by jaceklaskowski** [comment]
I don't agree with @andrewor14. It does add a value of being consistent with how Spark informs about its status - if it says "Stopping..." at INFO it should be corresponding "Starting..." at INFO. That was my initial goal. Consistency is the goal and the value here.

Why does Spark report about services being stopped at all? What's the rationale?

I would change it to DEBUG, but since all the other logs about starting services are at INFO, I'd leave it as is until someone reports it should be done.

**Comment by xguo27** [comment]
@andrewor14 What is your take on Jacek's comment? I don't think it's a bad idea to make it more consistent with a matching log message. Please let me know. Thx!

**Comment by andrewor14** [comment]
I just don't see any value. It's obvious that the scheduler is starting / stopping. I just can't imagine anyone finding this message useful.

**Comment by markhamstra** [comment]
Beyond that, the message is actually somewhat misleading.  The "Stopping" message occurs in `stop()`, which is responsible for stopping the messageScheduler, eventProcessLoop and taskScheduler.  When the "Starting" message is logged, at least the eventProcessLoop and taskScheduler will already have been started, so the argument from symmetry for "Starting" because there is a "Stopping" doesn't hold even without considering that "Starting", ...logMessage, ...logMessage, ... isn't symmetrical to logMessage, ...logMessage, "Stopping" unless we can somehow run time both forwards and backwards.  There is also the problem that the "Starting" message is only sent from the secondary constructors and not from the primary.

**Comment by rxin** [comment]
I think this message is complete useless. I'd just remove it rather than spending time bickering about the wording.

**Comment by xguo27** [comment]
OK, I will close it. Thanks!

**Pull request closed without merging**